### PR TITLE
SWAT-305: Agent installation additional validation.

### PR DESCRIPTION
**Pull request by valdislav**

### CODE REVIEW

**Review by balex13 — APPROVED** [review]

File: install.sh [hunk 1]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
-[ "$installDaemon" ] && installAndConfigureDaemon || installAndConfigureCron
+echo "** Install validation "
+
+if [[ ! -f "$appRoot/app/etc/env.php" ]]; then
+  error_exit "Magento not found."
+else
+  echo "Magento Found - OK"
+fi
+
+if nc -z $updaterDomain 443 2>/dev/null; then
```
Comment (yvoronoy) [inline install.sh]: netcat must be a dependency. 

1. See function _checkDependencies_
2. Update https://devdocs.magento.com/tools/site-wide-analysis.html

File: install.sh [hunk 2]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
+    echo "Connect to API Server - OK"
+else
+    error_exit "Can not connect to API Server"
+fi
+
+if [ -f "$agentPath/config.yaml" ]; then
+    echo "Config File is created - OK"
+else
+    error_exit "Config File was not created."
```
Comment (yvoronoy) [inline install.sh]: Lets increase visibility to simplify troubleshooting
```
error_exit "Config File $agentPath/config.yaml was not created."
```

File: install.sh [hunk 3]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
+if [[ ! -f "$appRoot/app/etc/env.php" ]]; then
+  error_exit "Magento not found."
+else
+  echo "Magento Found - OK"
+fi
+
+if nc -z $updaterDomain 443 2>/dev/null; then
+    echo "Connect to API Server - OK"
+else
+    error_exit "Can not connect to API Server"
```
Comment (yvoronoy) [inline install.sh]: Increase visibility and simplify troubleshooting

```
error_exit "Can not connect to API Server $updaterDomain and port 443"

File: install.sh [hunk 4]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
+fi
+
+phpVersion=$($phpPath -v | awk '{ print $2 }' | head -1)
+semver=( ${phpVersion//./ } )
+major="${semver[0]}"
+minor="${semver[1]}"
+
+echo "**Checking php version."
+
+if [ "$major" -eq 7 ]; then
```
Comment (yvoronoy) [inline install.sh]: As far I heard we going to support PHP8

File: install.sh [hunk 5]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
+        error_exit "php engine reachable by $phpPath is $phpVersion and is not supported."
+    fi
+else
+    echo "php version - OK"
+fi
+
+mkdir $agentPath/tmp
+
+if [ -w "$agentPath/tmp" ] ; then
+  echo "Temporary Folder is writeable - OK"
```
Comment (yvoronoy) [inline install.sh]:
```
echo "Temporary Folder $agentPath/tmp is writeable - OK"
```

File: install.sh [hunk 6]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
+else
+  error_exit "Temporary Folder in agent directory is not writable"
```
Comment (yvoronoy) [inline install.sh]: error_exit "Temporary Folder $agentPath/tmp in agent directory is not writable"

File: install.sh [hunk 7]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
...
+fi
+firstRun=$("$agentPath/scheduler")
+if [[ "$firstRun" == *"is going to update"* ]]; then
+  printSuccess "The Site Wide Analysis Tool Agent has been successfully installed at $agentPath"
+  [ "$installDaemon" ] && installAndConfigureDaemon || installAndConfigureCron
+else
+  error_exit "Failed to update launcher"
```
Comment (yvoronoy) [inline install.sh]:
```
error_exit "Failed to update launcher"
```
Please give a more guidance, what needs to be done next?
 - Start over again from the scratch
 - Check the logs

File: install.sh [hunk 8]
```diff
@@ -167,5 +167,55 @@ runchecksonstart: true
 loglevel: error
 EOF
 
-printSuccess "The Site Wide Analysis Tool Agent has been successfully installed at $agentPath"
-[ "$installDaemon" ] && installAndConfigureDaemon || installAndConfigureCron
+echo "** Install validation "
+
+if [[ ! -f "$appRoot/app/etc/env.php" ]]; then
+  error_exit "Magento not found."
```
Comment (yvoronoy) [inline install.sh]: Increase message verbosity

error_exit "Magento not found in the path $appRoot"

**Review by yvoronoy — CHANGES_REQUESTED** [review]

File: install.sh (hunk 4 above)
Comment (valdislav) [reply install.sh]: It will not fail, this condition is specifically for php7

**Review by yvoronoy — CHANGES_REQUESTED** [review]
Script failed 
```
** Install validation
Magento Found - OK
Connect to API Server - OK
Config File is created - OK
./install.sh: line 193: semver[1]: unbound variable
```

**Comment by yvoronoy** [comment]
Functional tests are added, @valdislav please merge main branch

File: install.sh (hunk 1 above)
Comment (valdislav) [reply install.sh]: I think there is no system without netcat, are you sure we should list it?

Comment (valdislav) [reply install.sh]: I added PR to documentation https://github.com/magento/devdocs/pull/9391

Comment (yvoronoy) [reply install.sh]: @valdislav please use the following code to replace nc dependency
```
if wget --timeout 5 --connect-timeout=5  --tries=1 --spider --verbose https://updater.swat.magento.com:443 2>&1 > /dev/null | grep -q "connected"
then
  echo "Connect to API Server - OK"
else
  error_exit "Can not connect to API Server $updaterDomain and port 443"
fi
```

**Pull request merged**